unitex/src/fr/umlv/unitex/leximir/delac/EditorDelac.java

```diff
@@ -0,0 +1,1292 @@
+                dataForSynSem2.put(pos, new HashMap<String, HashMap<String, String>>());
+                dataForSynSem2.get(pos).put(realSynSem, new HashMap<String, String>());
+                dataForSynSem2.get(pos).get(realSynSem).put(domainCategory, "1");
+            } else {
+                if (!dataForSynSem2.get(pos).containsKey(realSynSem)) {
+                    dataForSynSem2.get(pos).put(realSynSem, new HashMap<String, String>());
+                    dataForSynSem2.get(pos).get(realSynSem).put(domainCategory, "1");
+                } else {
+                    if (!dataForSynSem2.get(pos).get(realSynSem).containsKey(domainCategory)) {
+                        dataForSynSem2.get(pos).get(realSynSem).put(domainCategory, "1");
+                    } else {
+                        int count = Integer.parseInt(dataForSynSem2.get(pos).get(realSynSem).get(domainCategory)) + 1;
+                        if (dataForSynSem2.get(pos).get(realSynSem).containsKey(domainCategory)) {
+                            dataForSynSem2.get(pos).get(realSynSem).put(domainCategory, String.valueOf(count));
+                        }
+                    }
+                }
+            }
+        }
+
+        Map<String, Object[]> statSimSem = new HashMap<>();
+        int v = 2;
+        for (Map.Entry<String, HashMap<String, HashMap<String, String>>> t : dataForSynSem2.entrySet()) {
+            String key = t.getKey();
+            for (Map.Entry<String, HashMap<String, String>> y : t.getValue().entrySet()) {
+                for (Map.Entry<String, String> u : y.getValue().entrySet()) {
+                    statSimSem.put(String.valueOf(v), new Object[]{key, y.getKey(), u.getKey(), u.getValue()});
+                    v++;
+                }
+            }
+        }
+
+        GlobalProjectManager.search(null).getFrameManagerAs(UnitexInternalFrameManager.class)
+                .newStatisticOutput(statSimSem);
+
+    }
+
+    @SuppressWarnings("unchecked")
+    private void jTextFieldPosKeyPressed(java.awt.event.KeyEvent evt) {
+        if (evt.getKeyCode() == KeyEvent.VK_ENTER) {
+            JTextField textField = (JTextField) evt.getSource();
+            String text = textField.getText();
+            TableRowSorter<DefaultTableModel> rowSorter;
+            rowSorter = new TableRowSorter<>(tableModel);
+            this.getjTable1().setRowSorter(rowSorter);
+            this.getjTable1().removeAll();
+            if (text.trim().length() == 0) {
+                rowSorter.setRowFilter(null);
+            } else {
+                if (jCheckBoxExtract.isSelected()) {
+                    text = "^" + text + "$";
+                } else {
+                    if (!text.contains(".") && !text.contains("$")) {
+                        text = "^" + text;
+                    }
+                }
+                RowFilter rowFilter = RowFilter.regexFilter("(?i)" + text, 0);// search column at index 0
```

Comment:
Is there a reason why RowFilter is not parametized ?
If it was it would allow to remove `@SuppressWarnings("unchecked")`